core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1140,6 +1197,10 @@ public TableMetadata build() {
       );
     }
 
+    public SnapshotRef ref(String ref) {
```

Comment:
This can be package-private, right?

Reply:
For now yes. But I think for table scan/updating retention logic, we'll need it then. We can expose this when we get there?

Reply:
I don't think those cases are going to use the builder. I think we should keep this limited for now. We can always expose more, but it's hard to go the other way.